### ISSUE TITLE
Warning message in DHARMa

### ISSUE THREAD

**Issue by klattbk** [florianhartig/DHARMa#84]
Hi,

At first thanks for the brilliant DHARMa package, which makes my life so much easier!

Concerning my problem: when I am trying to get simulated residuals from a glmmTMB model with normally distributed data and with/without accounting for zero-inflation (ziformula), I am getting the following warning message: 

> res=simulateResiduals(m0)
Model family was recoginzed or set as continous, but duplicate values were detected in the simulation - changing to integer residuals (see help for details)
> plot(res)
> 

I tried to find something in the help section but did not succeed. Could you maybe give me a hint?

Thanks in advance!

Björn

**Comment by florianhartig** [comment]
Hi Björn,

this is a correct behavior of the package, but I guess I should document this better. 

The issue is that DHARMa needs to decide if the modeled regression uses a continuous or discrete distribution (because in case of discrete distribution, there is an additional smoothing performed, see vignette). 

Now, the model has a normal distribution, which DHARMa recognizes as continuous. However, there are bunch of cases, such as added zero-inflation or some parameters of the tweedie distribution, where discrete values (in particular lots of zeros occur). 

Because it's hard to know from the model settings if this is the case of not, I check in the outputs if values are repeated. If so, the distribution is switched to discrete. 

In short: in your case, DHARMa recognizes that, due to the zero inflation, it should treat your distribution as discrete, although it's a normal (which is what it should do), and this is what the message tells you!

**Comment by klattbk** [comment]
Hi Florian,

Thanks for your fast help and the nice explanation!
